src/main/java/org/springframework/hateoas/UriTemplate.java

```diff
@@ -203,6 +203,28 @@ public URI expand(Object... parameters) {
 		return builder.build().toUri();
 	}
 
+	/**
+	 * Expands the {@link UriTemplate} using the given parameters. The values will be applied in the order of the
+	 * variables discovered.
+	 *
+	 * @param parameters
+	 * @return
+	 * @see #expand(Map)
+	 */
+	public URI checkParams(Object... parameters) {
```

Comment:
I am not sure I get the purpose of this method. It'll always return `URI.create(baseUri)`. So what is all the code before supposed to do? The method name is quite misleading to. There's no check going on anywhere. It just returns the base URI. If that alone is worth doing, it could very well be added to `expand(…)` directly.

Reply:
I need to add the info that I am using the 0.25.0 version of haetoas and as so a 4.3 version of spring-web. In master of spring-web I can see that the encoding has changed and a enum for the encoding state has been added. Maybe this fixes the problem in spring 5.